cpr/session.cpp

```diff
@@ -47,13 +48,15 @@ class Session::Impl {
     Url url_;
     Parameters parameters_;
     Proxies proxies_;
+    struct curl_httppost* formpost_;
```

Comment:
I don't think this is a good idea. You're placing here a raw pointer that is owning.

This demands you add a destructor, which in turn demands you add custom copy and move semantics. I think it's more code than you have to. Place the pointer in a unique_ptr, initialize it with the `curl_formfree()` function, and then you don't need any of the destructor, copy and move functions. You also don't need the check for null in the cleanup, because that's already done in `curl_formfree()`.

I haven't tried coding this, but I imagine that when you do, you'll see that `Session` loses its copy semantics, because `std::unique_ptr<>` doesn't allow copying. This in turn won't compile if `Session` needs copy semantics. Does it? Either way, this is a great way to find a bug at compile time. Because this PR doesn't add the (otherwise) necessary copy semantics, so if you'd copy a `Session` now, you'll have a double-free, and that's UB.